### ISSUE TITLE
SearchBuilder Requires DateTime when used with dates.

### ISSUE THREAD

**Issue by pali07** [mwouts/itables#288]
![image](https://github.com/mwouts/itables/assets/124549717/e893c27e-738d-41b4-bc4d-522667056190)

Can anyone please help me with this issue. unable to install the datetime library

**Comment by mwouts** [comment]
Hi @pali07 , thanks for reporting this.

Could you show us the error message, and copy your example as text too? Thanks!

**Comment by pali07** [comment]
import pandas as pd
from itables import init_notebook_mode, show
from datetime import date, datetime

# Initialize itables for notebook display
init_notebook_mode(all_interactive=True)

def convert_to_table(data, description=None, column_names=None, specified_columns=None):

    df = pd.DataFrame(data, columns=column_names)
    if column_names is None and df.columns.nlevels == 1:
        column_names = df.columns.tolist()

    if column_names:
        column_names = [col.strip('_').capitalize() for col in column_names]
        df.columns = column_names

    if specified_columns:
        df = df[specified_columns]

    if description:
        # Dummy function for description formatting; replace with actual implementation
        def format_description(desc):
            return desc

        description = format_description(description)
        caption = f'<caption style="caption-side: top; margin-bottom: 10px;">{description}</caption>'
    else:
        caption = '<caption style="caption-side: top;font-weight: bold">Tabular format</caption>'

    # Display the DataFrame using itables
    show(
        df,
        tags=caption,
        classes="display nowrap cell-border top-border",
        scrollX=True,
        dom="Bfrtip",
        buttons=[
            "colvis",
            "pageLength",
            {
                "extend": "collection",
                "text": "Export",
                "buttons": [
                    {"extend": "csvHtml5", "title": "data"},
                    {"extend": "excelHtml5", "title": "data"}
                ]
            },
            "searchPanes",
            "searchBuilder"
        ],
        select=True
    )

# Example data
data = {
    'Title': ['1Password', '1Password', '1Password'],
    'Name': [12296, 24812, 25516],
    'AnyDateColumn': ['2023-04-19', '2023-04-18', '2023-04-18']
}

convert_to_table(data)

**Comment by pali07** [comment]
Not getting any error message getting table but search builder is not working
![image](https://github.com/mwouts/itables/assets/124549717/d175d4a1-1afa-46b9-9524-3b92c53c5318)

After clicking on searchbuilder button the output is as follows 

![image](https://github.com/mwouts/itables/assets/124549717/072038d2-b656-45c3-9e50-2b99dd795f9d)

**Comment by pali07** [comment]
```commandline
import pandas as pd
from itables import init_notebook_mode, show
from datetime import date, datetime

init_notebook_mode(all_interactive=True)

def convert_to_table(data, description=None, column_names=None, specified_columns=None):

    df = pd.DataFrame(data, columns=column_names)
    if column_names is None and df.columns.nlevels == 1:
        column_names = df.columns.tolist()
    if column_names:
        column_names = [col.strip('_').capitalize() for col in column_names]
        df.columns = column_names

    if specified_columns:
        df = df[specified_columns]

    show(
        df,
        classes="display nowrap cell-border top-border",
        scrollX=True,
        dom="Bfrtip",
        buttons=[
            "colvis",
            "pageLength",
            {
                "extend": "collection",
                "text": "Export",
                "buttons": [
                    {"extend": "csvHtml5", "title": "data"},
                    {"extend": "excelHtml5", "title": "data"}
                ]
            },
            "searchPanes",
            "searchBuilder"
        ],
        select=True
    )
data = {
    'Title': ['1Password', '1Password', '1Password'],
    'Name': [12296, 24812, 25516],
    'AnyDateColumn': ['2023-04-19', '2023-04-18', '2023-04-18']
}

convert_to_table(data)

```

**Comment by pali07** [comment]
Hi @mwouts  Thanks for the quick reply

**Comment by mwouts** [comment]
Thanks for the detailed information! I can reproduce the issue, that's a first step. I think we should be able to add the datetime extension, cf. the linked PR, although it's not working right now...

At the moment I am getting the following error: `Uncaught ReferenceError: assignment to undeclared variable dt` (and no table can be displayed). I might have to see on the datatables forum for how to get this PR right. I will keep you posted.

**Comment by rsnell-usgs** [comment]
+1 for this issue. Same error for a table that includes dates as a column.

**Comment by mwouts** [comment]
Oh that was helpful! I followed the first comment that points to the [download builder](https://datatables.net/download/) and I could install the DateTime package.

@rsnell-usgs @pali07 feel free to use the install instructions at https://github.com/mwouts/itables/pull/289#issuecomment-2159267444 to test the PR locally.

I am not going to merge this right now as I still see a small issue, I will discuss it with Allan on the PR.

**Comment by mwouts** [comment]
This issue has been solved in `itables==2.1.5`
